### ISSUE TITLE
add diagnostic for Duplicated lfn name when inserting files.

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#6261]
the only publication errors  seen in first days of operation of new publisher are due to DBS exception
  Duplicated lfn name when inserting files.
There are no duplicates in the list and the code already check against the already published files.

Indeed in the first case I investigated and the file was actually present in a different dataset, the
user could not find a reason for this and a subtle bug is suspected  see #6252 

If it keeps happening, I need to investigate. But tracking things from the dumped failed block is a pain and slow,
it would be better that when TaskPublish hits this exception, it does a lookup in DBS for all the LFN's it is trying to insert to find out which one(s) is giving the problem, and prints out what it finds.

**Comment by belforte** [comment]
keep happening, although at low rate. Last occurrence on Nov 21 in
`logs/tasks/vcepaiti/201118_182833:vcepaiti_crab_QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016_NANOX_201117.log`

But I see that the datasetname was wront with the `00000000000000000000000000000000` instead of the pset hash like other runnings of the script had.
SO maybe this problem originates from some bad data inserted in the DB by the postjob ?

```
2020-11-21 01:45:29,573:INFO: Start new iteration. Get files to publish                                                                                                                                                 │
2020-11-21 01:45:29,582:INFO: Will connect to CRAB service: other                                                                                                                                                       │
2020-11-21 01:45:29,583:INFO: Will use restHost and dbInstance from config file                                                                                                                                         │
2020-11-21 01:45:29,583:INFO: Will connect to CRAB Data Base via URL: https://cmsweb-k8s-prod.cern.ch//crabserver/prod                                                                                                  │
2020-11-21 01:45:29,760:INFO: DBS Source API URL: https://cmsweb.cern.ch/dbs/prod/global/DBSReader                                                                                                                      │
2020-11-21 01:45:29,760:INFO: DBS Global API URL: https://cmsweb.cern.ch/dbs/prod/global/DBSReader                                                                                                                      │
2020-11-21 01:45:29,760:INFO: DBS Destination API URL: https://cmsweb.cern.ch/dbs/prod/phys03/DBSWriter                                                                                                                 │
2020-11-21 01:45:29,760:INFO: DBS Destination read API URL: https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader                                                                                                            │
2020-11-21 01:45:29,761:INFO: DBS Migration API URL: https://cmsweb.cern.ch/dbs/prod/phys03/DBSMigrate                                                                                                                  │
2020-11-21 01:45:29,761:INFO: inputDataset: /QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8/RunIISummer16MiniAODv3-PUMoriond17_94X_mcRun2_asymptotic_v3-v2/MINIAODSIM                                             │
2020-11-21 01:45:29,962:INFO: Successfully inserted primary dataset QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8.                                                                                               │
2020-11-21 01:45:29,983:INFO: Dataset /QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8/vcepaiti-QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016-00000000000000000000000000000000/USER already contains 0 │
2020-11-21 01:45:29,983:INFO: Published output config.                                                                                                                                                                  │
2020-11-21 01:45:29,983:INFO: About to insert dataset: {'dataset': u'/QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8/vcepaiti-QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016-00000000000000000000000000│
2020-11-21 01:45:30,229:INFO: Found 11 files not already present in DBS which will be published.                                                                                                                        │
2020-11-21 01:45:30,385:ERROR: Error for files: [u'/store/user/vcepaiti/HNL/NANOX_201117/QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016/QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8/QCD_Pt-80to120_E│
2020-11-21 01:45:30,387:ERROR: Error when publishing (/store/temp/user/vcepaiti.d1821f021e371deb2f4bc240a1303dd7a1edee19/HNL/NANOX_201117/QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016/QCD_Pt-80to120_EMEnr┤
  File "/build/cmsbld/jenkins/workspace/CRABServer_BuildOnRelease/w/BUILD/slc7_amd64_gcc630/cms/crabtaskworker/v3.201117/CRABServer-v3.201117/build/lib/Publisher/TaskPublish.py", line 837, in publishInDBS3           ┤
  File "/data/srv/TaskManager/v3.201117/slc7_amd64_gcc630/cms/dbs3-client/3.15.1/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 346, in insertBulkBlock                                                       │
    result =  self.__callServer("bulkblocks", data=blockDump, callmethod='POST' )                                                                                                                                       │
  File "/data/srv/TaskManager/v3.201117/slc7_amd64_gcc630/cms/dbs3-client/3.15.1/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 203, in __callServer                                                          │
    self.__parseForException(http_error)                                                                                                                                                                                │
  File "/data/srv/TaskManager/v3.201117/slc7_amd64_gcc630/cms/dbs3-client/3.15.1/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 230, in __parseForException                                                   │
    raise HTTPError(http_error.url, data['exception'], data['message'], http_error.header, http_error.body)                                                                                                             │
HTTPError: HTTP Error 400: Duplicated lfn name when inserting                    files.                                                                                                                                 │
                                                                                                                                                                                                                        │
2020-11-21 01:45:31,797:ERROR: FAILING BLOCK DUE TO HTTP Error 400: Duplicated lfn name when inserting                    files.  SAVED AS /data/srv/Publisher_files/FailedBlocks/failed-block-at-1605919530.39.txt     │
2020-11-21 01:45:31,797:INFO: End of publication status for dataset /QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8/vcepaiti-QCD_Pt-80to120_EMEnriched_TuneCUETP8M1_13TeV_pythia8-2016-000000000000000000000000000│
2020-11-21 01:45:31,797:INFO: Marking 11 file(s) as failed                                                                                                                                                              │
2020-11-21 01:45:32,678:INFO: marked 10 files                                                                                        
```

OTOH failed blocks dumped up to Nov 20 have a valid looking pset hash
```
[crab3@crab-prod-tw02 Publisher_schedd]$ grep -l 00000000 PublisherFiles/FailedBlocks/failed-block-at*
PublisherFiles/FailedBlocks/failed-block-at-1605919530.39.txt
[crab3@crab-prod-tw02 Publisher_schedd]$ 
```

so this particular thing on Nov 21 was "special".

**Comment by belforte** [comment]
set prio=top to do it now.

**Comment by belforte** [comment]
After checking blocks which failed to publish so far, I found 3 categories:
1. a bunch of failures on Nov 17 when we migrated... to be ignored, many things have improved since
2. a bunch of failures in Dec 17 where the failing block contains files which are almost all already present in the target dataset. This makes no sense since there is a check in the code already. Maybe at some point there were two publishers running or some other temporary confusion. Not worth adding ad-hoc code, and anyhow if files are are already published, user is happy
3. one recent failure on Jan 21 where the crab task has two output datasets which is known not to work with new publisher and one file inside the block was actually belonging to the wrong dataset.  We should avoid submitting such tasks until it is fixed. Item for CRABClient.

In conclusion, I see no need to make Publisher code more complex, also digging error messages from log files still is a nuisance. So I am simply adding to the repository a new script which I used to diagnose those use case `scripts/Utils/DebugFailedBlockPublication.py` which can be used on block dumps in `/data/srv/Publisher_files/FailedBlocks` and removed old failures from there.

**Comment by belforte** [comment]
I have placed instructions and one example in the twiki:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/Crab3OperatorDebugging#Debug_failed_publications
